src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -88,10 +88,12 @@ public HostOutputMessage HostOutputMessage
                 ParentId = functionStartedMessage.ParentId,
                 FunctionName = functionStartedMessage.Function.ShortName,
                 TriggerReason = functionStartedMessage.ReasonDetails,
-                StartTime = functionStartedMessage.StartTime.DateTime
+                StartTime = functionStartedMessage.StartTime.DateTime,
+                Properties = new Dictionary<string, object>()
             };
 
-            Stopwatch sw = Stopwatch.StartNew();
+            await this.NotifyPreBindAsync(fastItem);
```

Comment:
Yes. This is the key part. Subtle but critical. SDK already exposes an instrumentation stream, and this is adding another notification. 

---
In reply to: [126777925](https://github.com/Azure/azure-webjobs-sdk/pull/1218#discussion_r126777925) [](ancestors = 126777925)